core/vm/actor/storage_power_actor.cpp

```diff
@@ -0,0 +1,242 @@
+    return outcome::success();
+  }
+
+  bool StoragePowerActor::minerNominalPowerMeetsConsensusMinimum(
+      const power::Power &miner_power) {
+    // if miner is larger than min power requirement, we're set
+    if (miner_power >= kMinMinerSizeStor) {
+      return true;
+    }
+
+    // otherwise, if another miner meets min power requirement, return false
+    if (num_miners_meeting_min_power > 0) {
+      return false;
+    }
+
+    // else if none do, check whether in MIN_MINER_SIZE_TARG miners
+    if (power_table_->getSize() <= kMinMinerSizeTarg) {
+      // miner should pass
+      return true;
+    }
+
+    // get size of MIN_MINER_SIZE_TARGth largest miner
+    return miner_power >= power_table_->getMaxPower();
+  }
+
+  outcome::result<void> StoragePowerActor::setClaimedPowerEntryInternal(
+      const primitives::address::Address &miner_addr,
+      const power::Power &updated_power) {
+    return claimed_power_->setMinerPower(miner_addr, updated_power);
+  }
+  outcome::result<power::Power> StoragePowerActor::getPowerTotalForMiner(
+      const primitives::address::Address &miner_addr) const {
+    return power_table_->getMinerPower(miner_addr);
+  }
+
+  outcome::result<power::Power> StoragePowerActor::getNominalPowerForMiner(
+      const primitives::address::Address &miner_addr) const {
+    return nominal_power_->getMinerPower(miner_addr);
+  }
+
+  outcome::result<power::Power> StoragePowerActor::getClaimedPowerForMiner(
+      const primitives::address::Address &miner_addr) const {
+    return claimed_power_->getMinerPower(miner_addr);
+  }
+
+  outcome::result<void> StoragePowerActor::addMiner(
+      const primitives::address::Address &miner_addr) {
+    auto check = power_table_->getMinerPower(miner_addr);
+    if (!check.has_error()) return ALREADY_EXIST;
+
+    OUTCOME_TRY(power_table_->setMinerPower(miner_addr, 0));
+    OUTCOME_TRY(nominal_power_->setMinerPower(miner_addr, 0));
+    OUTCOME_TRY(claimed_power_->setMinerPower(miner_addr, 0));
+    return outcome::success();
+  }
+
+  outcome::result<void> StoragePowerActor::removeMiner(
+      const primitives::address::Address &miner_addr) {
+    OUTCOME_TRY(power_table_->removeMiner(miner_addr));
+    OUTCOME_TRY(nominal_power_->removeMiner(miner_addr));
+    OUTCOME_TRY(claimed_power_->removeMiner(miner_addr));
+    auto position = std::find(po_st_detected_fault_miners_.cbegin(),
+                              po_st_detected_fault_miners_.cend(),
+                              miner_addr);
+    if (position != po_st_detected_fault_miners_.cend())
+      po_st_detected_fault_miners_.erase(position);
+    return outcome::success();
+  }
+
+  StoragePowerActor::StoragePowerActor(
+      std::shared_ptr<Indices> indices,
+      std::shared_ptr<crypto::randomness::RandomnessProvider>
+          randomness_provider)
+      : indices_(std::move(indices)),
+        randomness_provider_(std::move(randomness_provider)) {
+    total_network_power_ = 0;
```

Comment:
M-m-m... Why not initialize everything in the initializer list?

Reply:
Fixed